api/server/router/network/network_routes.go

```diff
@@ -55,7 +56,18 @@ SKIP:
 				continue SKIP
 			}
 		}
-		list = append(list, *n.buildNetworkResource(nw))
+
+		var nr *types.NetworkResource
+		// Versions < 1.26 fetches all the containers attached to a network
+		// in a network list api call. It is a heavy weight operation when
+		// run across all the networks. Starting API version 1.26, this detailed
+		// info is available for network specific GET API (equivalent to inspect)
+		if versions.LessThan(httputils.VersionFromContext(ctx), "1.26") {
+			nr = n.buildDetailedNetworkResources(nw)
+		} else {
+			nr = n.buildNetworkResource(nw)
+		}
+		list = append(list, *nr)
 	}
 
 	list, err = filterNetworks(list, netFilters)
```

Comment:
As a follow up, it may be worth considering to add information _not_ needed for filtering until after the networks have been filtered.

Perhaps create utility functions to append information such as `containers` and `peers` (e.g. `appendEndpointInfo(list)`, `appendPeerInfo(list)`)